src/http.rs

```diff
@@ -205,7 +205,16 @@ pub fn build_tls_connector(
     let settings = tls_settings.tls().cloned();
     https.set_callback(move |c, _uri| {
         if let Some(settings) = &settings {
-            settings.apply_connect_configuration(c);
+            match settings.apply_connect_configuration(c) {
+                Ok(()) => (),
+                Err(error) => {
+                    error
+                        .source()
+                        .unwrap()
+                        .downcast_ref::<openssl::error::ErrorStack>()
+                        .unwrap();
```

Comment:
Rather than `unwrap` could we use `expect`? It's not immediately clear to me why these unwraps are safe and I think `expect` would make that clearer since you can provide a reason.

Reply:
changed it to expect and provided reason why it is expected.